src/nope.ts

```diff
@@ -0,0 +1,61 @@
+import { appendErrors, transformToNestObject, Resolver } from 'react-hook-form';
+import NopeObject from 'nope-validator/lib/umd/NopeObject';
+
+const parseNopeSchema = (error: any, validateAllFieldCriteria: boolean) => {
+  const errors = Object.keys(error);
+
+  return Array.isArray(errors)
+    ? errors.reduce((acc: Record<string, any>, fieldName) => {
+        return {
+          ...acc,
+          ...(fieldName
+            ? acc[fieldName] && validateAllFieldCriteria
```

Comment:
I think you should test here too. (the case `validateAllFieldCriteria` is false)